### PR TITLE
Support a maximum for number of buffers shown in mode-line and a flag whether highlighted buffers should be shown first.

### PR DESCRIPTION
**Pull request by ecraven**
Add custom variable `tracking-max-mode-line-entries'.
If set to an integer, this limits the number of non-highlighted buffers shown in the mode-line.

track-max-buffers

I am lurking in some 30 channels, so this helps with the problem that the mode-line can become cluttered with a lot of activity.

Second commit contains `tracking-with-face-first'. If it is non-nil, all buffers with any highlight face will be shown before all buffers with no highlight face.

### CODE REVIEW

**Comment by ecraven** [comment]
CI failure:
````
Dependency buttercup failed to install: Wrong type argument: sequencep, buttercup
````
I suspect this is not a problem I introduced, but rather some problem with current melpa / buttercup?

File: tracking.el [hunk 1]
```diff
@@ -315,26 +323,44 @@ This returns a list suitable for `mode-line-format'."
...
-                mouse-face mode-line-highlight
-                help-echo ,(format (concat "New activity in %s\n"
-                                           "mouse-1: pop to the buffer")
-                                   (car buffer-names)))
-              result)
-        (setq buffer-names (cdr buffer-names)
-              shortened-names (cdr shortened-names))
-        (when buffer-names
-          (push "," result)))
+        (let* ((highlighted-p (eq 'circe-highlight-nick-face (get-text-property 0 'face (car buffer-names))))
```
Comment (wasamasa) [inline tracking.el]: You don't need `let*` here, do you?

**Comment by ecraven** [comment]
@wasamasa sorry, that was left over from a previous version, changed to =let=.

**Comment by retroj** [comment]
I see this as two features, not one - highlighted buffers first, and limiting the number of buffers shown - each of these things could be useful without the other.

**Comment by ecraven** [comment]
> I see this as two features, not one - highlighted buffers first, and limiting the number of buffers shown
> each of these things could be useful without the other.

I tend to to agree. Ideally reordering the buffers would happen in the function that adds the buffers to `tracking-buffers`, and only showing n is as shown in my PR (though a bit simpler, as there is no need to sort first).
@wasama, shall I split this into two PRs? Is this something you even want to merge in general?

**Comment by wasamasa** [comment]
@ecraven While I could actually do that, I won't.  It's not up to me to decide because I don't own this code and am not its maintainer either.

**Comment by jorgenschaefer** [comment]
Splitting up the logic would be nice. No need to make this two PRs, but two commits would be great. Happy to merge this, then, though!

**Comment by ecraven** [comment]
@jorgenschaefer: rewritten and force-pushed ;)

**Comment by jorgenschaefer** [comment]
Hm. Rebasing ontop of master should allow this to build now.

**Comment by ecraven** [comment]
Done ;)
Is there any place where I should add documentation, apart from the (existing) docstrings?

**Comment by jorgenschaefer** [comment]
I don't think so. The wiki maybe? Circe used to have an info manual, but it was a pain to update and I don't use info anymore, so it got scrapped. If you have ideas, I'm all ears, but I'm more interested in docstrings and tests :-D

**Comment by jorgenschaefer** [comment]
Merged. Thank you very much for your work!

**Comment by wasamasa** [comment]
@ecraven I found a warning regarding `incf` when byte-compiling the code and pushed a fix to master, otherwise my message handlers would error out.

**Comment by ecraven** [comment]
@forcer: thanks for merging! I'll try to add to the wiki over the next few days.
@wasamasa: sorry about that, thanks for fixing it!

**Pull request merged**